File: leetcode 67.java
class Solution {
    public String addBinary(String a, String b) {
        StringBuilder sb = new StringBuilder();
        int i = a.length() - 1, j = b.length() - 1;
        int sum = 0;
      
        while(i >= 0 || j >= 0) {
            if(i >= 0) sum += a.charAt(i--) - '0';
            if(j >= 0) sum += b.charAt(j--) - '0';
            sb.append(sum % 2);
            sum = sum / 2;
        }
        if(sum != 0) sb.append(sum);
        return sb.reverse().toString();
    }

}

File: python version/leetcode 67.py
class Solution(object):
    def addBinary(self, a, b):
        """
        :type a: str
        :type b: str
        :rtype: str
        """
        i, j, cout = len(a)-1, len(b)-1, 0
        res = []
        while i >= 0 or j >= 0:
            if i >= 0: cout += int(a[i])
            if j >= 0: cout += int(b[j])
            i -= 1
            j -= 1
            res.append(cout%2)
            cout = cout // 2
            
        if cout != 0 : res.append(cout)
        return "".join(list(map(str,res[::-1]))) 